object/slicer/slicer.go

```diff
@@ -414,13 +416,18 @@ func flushObjectMetadata(signer neofscrypto.Signer, meta dynamicObjectMetadata,
 	return id, nil
 }
 
-func writeInMemObject(signer neofscrypto.Signer, w ObjectWriter, header object.Object, payload []byte, meta dynamicObjectMetadata) (oid.ID, error) {
+func writeInMemObject(signer neofscrypto.Signer, w ObjectWriter, header object.Object, payload []byte, meta dynamicObjectMetadata, session *session.Object) (oid.ID, error) {
 	id, err := flushObjectMetadata(signer, meta, &header)
 	if err != nil {
 		return id, err
 	}
 
-	stream, err := w.InitDataStream(header, signer)
+	var prm client.PrmObjectPutInit
+	if session != nil {
+		prm.WithinSession(*session)
+	}
+
+	stream, err := w.ObjectPutInit(context.TODO(), header, signer, prm)
```

Comment:
If we don't have any appropriate context here, it's a problem.

Reply:
Fixed

Reply:
Well, it's a problem. golang/go#20280